### DISCUSSION TITLE
Packages/com.unity.visualscripting/manual/vs-create-custom-node-add-ports

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#618] [Notes]
# Packages/com.unity.visualscripting/manual/vs-create-custom-node-add-ports

https://docs.unity3d.com/Packages/com.unity.visualscripting@latest/index.html?subfolder=%2Fmanual%2Fvs-create-custom-node-add-ports.html

**Comment by Thundernerd** [comment]
An important thing to know is that if you want to have the default property inspector for your input value ports, you **have** to specify a default value. 

Omitting the default value will get you a port that you can only assign a value from another node to, not a port with an input field behind it like shown in the image above.
